sklearn/model_selection/tests/test_split.py

```diff
@@ -984,6 +985,39 @@ def test_label_kfold():
                          next, LabelKFold(n_folds=3).split(X, y, labels))
 
 
+def test_homogeneous_time_series_cv():
+    X = [[1, 2], [3, 4], [5, 6], [7, 8], [9, 10], [11, 12], [13, 14]]
+
+    # Should fail if there are more folds than samples
+    assert_raises_regexp(ValueError, "Cannot have number of folds.*greater",
+                         next,
+                         HomogeneousTimeSeriesCV(n_folds=9).split(X))
+
+    homo_tscv = HomogeneousTimeSeriesCV(3)
+
+    # Manually check that Homogeneous Time Series CV preserves the data
+    # ordering on toy datasets
+    splits = homo_tscv.split(X[:-1])
+    train, test = next(splits)
+    assert_array_equal(train, [0, 1])
+    assert_array_equal(test, [2, 3])
+
+    train, test = next(splits)
+    assert_array_equal(train, [0, 1, 2, 3])
+    assert_array_equal(test, [4, 5])
+
+    splits = HomogeneousTimeSeriesCV(3).split(X)
+    train, test = next(splits)
+    assert_array_equal(train, [0, 1, 2])
+    assert_array_equal(test, [3, 4])
+
+    train, test = next(splits)
+    assert_array_equal(train, [0, 1, 2, 3, 4])
+    assert_array_equal(test, [5, 6])
+
+    # Check get_n_splits returns the number of folds - 1
+    assert_equal(2, homo_tscv.get_n_splits())
```

Comment:
Should assert `len(list(splits)))` equals `get_n_splits()`. That is our contract.